tests/components/zha/test_light.py

```diff
@@ -1921,3 +1924,55 @@ async def test_group_member_assume_state(
         await zha_gateway.async_remove_zigpy_group(zha_group.group_id)
         assert hass.states.get(group_entity_id) is None
         assert entity_registry.async_get(group_entity_id) is None
+
+
+@pytest.mark.parametrize(
+    ("restored_state", "expected_cm"),
+    [
+        (STATE_ON, ColorMode.XY),
+        (STATE_OFF, None),
+    ],
+)
+async def test_restore_light_state(
+    hass: HomeAssistant,
+    zigpy_device_mock,
+    core_rs,
+    zha_device_restored,
+    restored_state,
+    expected_cm,
+) -> None:
+    """Test ZHA light restores without throwing an error when attributes are None."""
+
+    attributes = {
+        "brightness": None,
+        "off_with_transition": None,
+        "off_brightness": None,
+        "color_mode": None,
+        "color_temp": None,
+        "xy_color": None,
+        "hs_color": None,
+        "effect": None,
+    }
+
+    entity_id = "light.fakemanufacturer_fakemodel_light"
+    core_rs(
+        entity_id,
+        state=restored_state,
+        attributes=attributes,
+    )
+    await async_mock_load_restore_state_from_storage(hass)
+
+    zigpy_device = zigpy_device_mock(LIGHT_COLOR)
+    zha_device = await zha_device_restored(zigpy_device)
+    entity_id = find_entity_id(Platform.LIGHT, zha_device, hass)
+
+    assert entity_id is not None
+    assert hass.states.get(entity_id).state == restored_state
+
+    for attr in attributes:
+        if attr == "color_mode":
```

Comment:
Please don't use if statements in test code. Maybe you can use snapshots from syrupy or compare it with a dict you put in

Reply:
Is this good? https://github.com/home-assistant/core/pull/102806/commits/cdab3db837c803244f53d8efca435505b92ba8a0